### PR TITLE
user mail server migration

### PR DESCRIPTION
**Pull request by iammiracle**

### CODE REVIEW

File: user_mail_common/res_users.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _remote_id(self):
     forward_address = fields.Text('Forward address', help='Comma separated list of mail addresses')
     forward_cp = fields.Boolean('Keep', help="Keep a local copy of forwarded messages")
     postfix_active = fields.Boolean('Postfix Active', default=False,)
-    postfix_alias_ids = fields.One2many('postfix.alias', 'user_id', string='Alias', copy=False, ondelete="cascade")
```
Comment (nils-vertel) [inline user_mail_common/res_users.py:76]: why do we remove ondelete="cascade" here?

File: user_mail_common/res_users.py [hunk 2]
```diff
@@ -164,16 +164,19 @@ def _remote_id(self):
     total_quota = fields.Integer(compute="_total_quota", string='All quota (MB)', help="Sum of all Users Quota in MB")
     catchall = fields.Char(compute='_catchall', string='Catchall', help="catchall mail address")
     domain = fields.Char(string='Domain', help="the internet domain for mail", compute='_get_domain',
-                         inverse='_set_domain', store=True, required=True)
```
Comment (nils-vertel) [inline user_mail_common/res_users.py:167]: Why do we remove store=True here?

File: user_mail_common/res_users.py [hunk 3]
```diff
@@ -164,16 +164,19 @@ def _remote_id(self):
...
     def _set_domain(self):
         self.env['ir.config_parameter'].set_param('mail.catchall.domain', self.domain)
     
+    @api.depends('name')
     def _get_domain(self):
         for rec in self:
-            rec.domain = rec.env['ir.config_parameter'].get_param(
-                'mail.catchall.domain')
+            if rec.name:
+                rec.domain = rec.env['ir.config_parameter'].get_param('mail.catchall.domain')
```
Comment (nils-vertel) [inline user_mail_common/res_users.py:177]: get_param can return a default value that can be False I think?

File: user_mail_server/res_users.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015- Vertel AB (<http://www.vertel.se>).
+#
+#    This progrupdateam is free software: you can redistribute it and/or modify
```
Comment (nils-vertel) [inline user_mail_server/res_users.py]: progrupdateam?

**Review by nils-vertel — CHANGES_REQUESTED** [review]
Some minor things to look in to

File: user_mail_common/res_users.py (hunk 1 above)
Comment (iammiracle) [reply user_mail_common/res_users.py:76]: I think this should be applied on the corresponding Many2one, which has been done.

File: user_mail_common/res_users.py (hunk 2 above)
Comment (iammiracle) [reply user_mail_common/res_users.py:167]: Well. I see no reason why it should be stored. Moreover for existing record, the compute method did not run with store=True

File: user_mail_common/res_users.py (hunk 3 above)
Comment (iammiracle) [reply user_mail_common/res_users.py:177]: Yes, even that parameter needs to be defined in the system parameters.

File: user_mail_server/res_users.py (hunk 4 above)
Comment (iammiracle) [reply user_mail_server/res_users.py]: oh, that was copied from a module. I did not see that. It can be removed.

**Review by nils-vertel — APPROVED** [review]

**Pull request merged**